Advanced-Recruit-Training/src/BatteryStateDisplayService.cpp

```diff
@@ -29,5 +29,14 @@ void BatteryStateDisplayService::handleBatteryDataReceived(const BatteryData& ba
         << " Current: " << batteryData.current
         << " Total Ah used: " << batteryStateOfChargeService_.totalAmpHoursUsed() << endl;
 
-    // TODO: Print out time until the battery is fully charged or depleted.
+    QString batTime = batteryStateOfChargeService_.timeWhenChargedOrDepleted().toString("hh:mm:ss.zzz");
+
+    if (batteryStateOfChargeService_.isCharging())
+    {
+        QTextStream(stdout) << "Time until full charge: " << batTime << endl;
+    }
+    else
+    {
+        QTextStream(stdout) << "Time until full depletion: " << batTime << endl;
+    }
```

Comment:
I would recommend editing this function so the time to charge/depletion is displayed on the same line as the data, that way it's less confusing and easier to look at end result.